toolbox/util/gcsClient.go

```diff
@@ -48,6 +50,23 @@ func NewGCSClient(bucket string) *GCSClient {
 	}
 }
 
+// Exists finds if an object already exists on GCS bucket
+func (gcs *GCSClient) Exists(obj string) (bool, error) {
```

Comment:
do we need this or can we derive this from the error received while reading?

Reply:
Go does not distinguish or allow extending `error` so to do it that way we have to hack it with string matching, which to me seems kind of sketchy since the error message is meant for human consumption and could change any minute from server side.